File: helfomat-api/src/main/java/de/helfenkannjeder/helfomat/api/organisation/event/OrganizationEditAddVolunteerEventDto.java
package de.helfenkannjeder.helfomat.api.organisation.event;

import de.helfenkannjeder.helfomat.api.organisation.VolunteerDto;
import de.helfenkannjeder.helfomat.core.organisation.OrganisationId;

/**
 * @author Valentin Zickner
 */
public class OrganizationEditAddVolunteerEventDto extends OrganizationEventDto {
    private int index;
    private VolunteerDto volunteer;

    OrganizationEditAddVolunteerEventDto() {
    }

    public OrganizationEditAddVolunteerEventDto(OrganisationId organisationId, int index, VolunteerDto volunteer) {
        super(organisationId);
        this.index = index;
        this.volunteer = volunteer;
    }

    public int getIndex() {
        return index;
    }

    public VolunteerDto getVolunteer() {
        return volunteer;
    }

    @Override
    public <T> T visit(OrganizationEventDtoVisitor<T> visitor) {
        return visitor.visit(this);
    }

}

File: helfomat-api/src/main/kotlin/de/helfenkannjeder/helfomat/api/organization/event/OrganizationEditAddVolunteerEventDto.kt
package de.helfenkannjeder.helfomat.api.organization.event

import de.helfenkannjeder.helfomat.api.organization.VolunteerDto
import de.helfenkannjeder.helfomat.core.organization.OrganizationId

/**
 * @author Valentin Zickner
 */
data class OrganizationEditAddVolunteerEventDto(
    override val organizationId: OrganizationId,
    val index: Int,
    val volunteer: VolunteerDto,
    override val eventApplicable: EventApplicability
) : OrganizationEventDto {

    override fun <T> visit(visitor: OrganizationEventDtoVisitor<T>): T {
        return visitor.visit(this)
    }

}